jwst/tweakreg/tweakreg_step.py

```diff
@@ -111,6 +115,24 @@ class TweakRegStep(Step):
     def process(self, input):
         use_custom_catalogs = self.use_custom_catalogs
 
+        if self.separation <= _SQRT2 * self.tolerance:
+            self.log.error(
+                "Parameter 'separation' must be larger than 'tolerance' by at "
+                "least a factor of sqrt(2) to avoid source confusion."
+            )
+            self.skip = True
```

Comment:
As I mentioned in https://github.com/spacetelescope/jwst/pull/8476#discussion_r1603952516
I was mistaken and this does not set `cal_step` to `SKIPPED` for the input models.

Reply:
I think we have to be VERY careful setting a step skip param to `True` in real time during step execution. A similar thing is/was done in the outlier_detection step, which had to be removed when it was discovered that it caused problems when the step is being called multiple times to operate on multiple sets of inputs when doing level-3 processing of some kinds of observations. If you're processing 20 different sets of data (e.g. data from 20 different sources in a NIRSpec MOS observation) and you set `skip=True` when processing source 5, the step will then be skipped for all remaining iterations, which we don't want. 

All that should be done in these instances is to set the step *status* keyword to "SKIPPED", but not touch the `skip` param itself.